plugins/org.eclipse.n4js.ui/src/org/eclipse/n4js/ui/outline/MetaTypeAwareComparator.java

```diff
@@ -0,0 +1,92 @@
+/**
+ * Copyright (c) 2017 NumberFour AG.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   NumberFour AG - Initial API and implementation
+ */
+package org.eclipse.n4js.ui.outline;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import org.eclipse.emf.ecore.EClass;
+import org.eclipse.n4js.ts.types.TypesPackage;
+import org.eclipse.xtext.ui.editor.outline.IOutlineNode;
+import org.eclipse.xtext.ui.editor.outline.actions.SortOutlineContribution;
+import org.eclipse.xtext.ui.editor.outline.impl.EObjectNode;
+import org.eclipse.xtext.ui.editor.outline.impl.OutlineFilterAndSorter.IComparator;
+import org.eclipse.xtext.ui.editor.preferences.IPreferenceStoreAccess;
+
+import com.google.inject.Inject;
+
+/**
+ *
+ */
+public class MetaTypeAwareComparator implements IComparator {
```

Comment:
What is the purpose of this type-aware sorting? As a user, when I toggle "A -> Z" sort button, I intuitively expect the list to be sorted alphabetically. But the order presented seems to be random now...